app/controllers/api/v1/projects_controller.rb

```diff
@@ -61,13 +62,17 @@ def activities
       def project_params
         raise TypeError unless params.require(:data).require(:type) == 'projects'
 
-        params.require(:data).require(:attributes).permit(:name, :visibility, :archived)
+        params.from_jsonapi.require(:project).permit(:name, :visibility, :archived, :project_folder_id)
```

Comment:
why do we need it here?

Reply:
As I mentioned in a meeting last week, `from_jsonapi` translates the raw json:api parameters into “classic” rails parameters, with a model top-level key and attributes in it’s object. The classic rails parameters are more conventional and easier to work with. As an added bonus, it handles relationships and parses them as `#{relationship_name}_id` or `#{relationship_name}_ids` for has_many relationships.

Reply:
We do not create relationships if specific controller exists for that type

Reply:
def jsonapi_params(type)
  raise TypeError unless params.require(:data).require(:type) == type

  params.require(:data).require(:attributes)
end

Reply:
How would you save the `belongs_to` associations then? 
If you check `json:api` [docs](https://jsonapi.org/format/#document-resource-object-attributes) says:
> Although has-one foreign keys (e.g. author_id) are often stored internally alongside other information to be represented in a resource object, these keys SHOULD NOT appear as attributes.

So we SHOULD NOT use the belongs_to fields in the request `attributes` object and should use the `relationships` object instead. 
It means we need a custom method in each controller, where we are assigning belongs_to attributes.
To translate this
```
{
    "data": {
        "type": "project_folders",
        "id": 1,
        "attributes": {
            
        },
        "relationships": {
            "parent_folder": {
                "data": { 
                    "type": "project_folders",
                    "id": 3
                }
            }
        }
    }
}
```

to this
```
{ project_folder: { name: 'MyUpdatedFolder', parent_folder_id: 2}
``` 

Only for this parent folder, we need to translate parameters with something like: 
```
params.permit(data: { attributes: %i(name) })[:data][:attributes]
          .merge(parent_folder_id:
                   params.permit(data: { relationships: {parent_folder: {data: %i(id)}}})['data']['relationships']['parent_folder']['data']['id'])

```
And a gem is doing exactly that. 

And I am not sure what is considered a trusted source? 
just a few facts about this gem: 
- MIT license
- 100% test coverage
- 34k downloads
- Last commit 20days ago..